cli/mender-artifact/copy.go

```diff
@@ -261,3 +275,44 @@ func parseCLIOptions(c *cli.Context) int {
 		return parseError
 	}
 }
+
+// createTmpFileWithPerm Takes a file, and creates a temp-file copy of the
+// current file, with the permissions given by perm.
+func createTmpFileWithPerm(f *os.File, perm os.FileMode) (string, error) {
+
+	td, err := ioutil.TempDir("", "mender-artifact-install")
+	if err != nil {
+		return "", err
+	}
+
+	tf, err := os.OpenFile(filepath.Join(td,
+		filepath.Base(f.Name())), os.O_CREATE|os.O_WRONLY, perm)
+	if err != nil {
+		return "", err
+	}
+
+	_, err = io.Copy(tf, f)
+	if err != nil {
+		return "", err
+	}
+
+	// override umask
+	err = tf.Chmod(perm)
+	if err != nil {
+		return "", err
+	}
+
+	name := tf.Name()
+	Log.Debugf("Tempfile name: %s\n", name)
+
+	s, _ := tf.Stat()
+	err = tf.Close()
+	if err != nil {
+		return "", err
+	}
+
+	Log.Debugf("The tempfile: %s with permissions: %v\norg-perm: %s\nmode: %s\n",
+		name, s, perm, s.Mode())
```

Comment:
What is `org-perm` here? Isn't it the desired `FileMode`? I think it is confusing to mix permissions and mode

Reply:
Hmm, yeh. True. Rewording